cylc/flow/exceptions.py

```diff
@@ -254,9 +254,17 @@ def __str__(self):
 
 class ClientError(CylcError):
 
-    def __init__(self, message: str, traceback: Optional[str] = None):
+    def __init__(
+        self,
+        message: str,
+        traceback: Optional[str] = None,
+        workflow: Optional[str] = None
```

Comment:
Might be a good idea to call this `workflow_id` as we also have the tokenised variant.

Reply:
Don't think it matters much for these exceptions. The more important one would be `WorkflowRuntimeClient`
https://github.com/cylc/cylc-flow/blob/af64ace2f078b7d656c83c09db483a531f43b98f/cylc/flow/network/client.py#L64-L66

however that's probably out of scope